BSP/Simulator/Src/BSP_SPI.c

```diff
@@ -40,6 +40,7 @@ void BSP_SPI_Write(spi_port_t port, uint8_t *txBuf, uint32_t txLen) {
         free(msg);
         Fault_BitMap |= Fault_CRC;
         EnterFaultState();
+        return ERROR;
```

Comment:
There in no point in this after `EnterFaultState()` since EnterFaultState never leaves itself. Also, the return value is based on whether a parameter is passed when the port is not one of the two defined. That functionality is not implemented in this c file.